bftengine/src/bftengine/DbCheckpointManager.cpp

```diff
@@ -115,14 +116,62 @@ void DbCheckpointManager::loadCheckpointDataFromPersistence() {
 }
 
 void DbCheckpointManager::init() {
-  // check if there is chkpt data in persistence
-  loadCheckpointDataFromPersistence();
-  // if there is a checkpoint created in database but entry corresponding to that is not found in persistence
-  // then probably, its a partially created checkpoint and we want to remove it
-  auto listOfCheckpointsCreated = dbClient_->getListOfCreatedCheckpoints();
-  for (auto& cp : listOfCheckpointsCreated) {
-    if (dbCheckptMetadata_.dbCheckPoints_.find(cp) == dbCheckptMetadata_.dbCheckPoints_.end()) {
-      removeCheckpoint(cp);
+  // In case of scale with restart we have to update dbcheckpoint metadata from file system
+  // if any dbcheckpoints are present in the file system.
+  if (lastCheckpointSeqNum_ == 0 && EpochManager::instance().getSelfEpochNumber() > 0) {
```

Comment:
to be more precise, we can also check if we have scale command logged and its epoch number is one less than the current epoch number. (may be in future PR)